client/my-sites/site-settings/form-general.jsx

```diff
@@ -170,28 +171,56 @@ class SiteSettingsFormGeneral extends Component {
 		} );
 	};
 
+	renderLanguagePickerNotice = () => {
+		const { fields, translate } = this.props;
+
+		const langId = get( fields, 'lang_id', '' );
+		const matches = /^error_(\w+)$/.exec( langId );
+
+		let notice;
+		switch ( matches && matches[ 1 ] ) {
+			case 'cap':
+				notice = translate(
+					'The Site Language setting is disabled due to insufficient permissions.'
+				);
+			case 'const':
+				notice = translate(
+					'The Site Language setting is disabled because your site has the WPLANG constant set.'
+				);
+		}
+
+		return notice && <Notice text={ notice } isCompact />;
+	};
+
 	languageOptions() {
 		const {
 			eventTracker,
 			fields,
 			isRequestingSettings,
 			onChangeField,
 			siteIsJetpack,
+			supportsLanguageSelection,
 			translate,
 		} = this.props;
-		if ( siteIsJetpack ) {
+		if ( ! supportsLanguageSelection ) {
 			return null;
 		}
+
 		return (
-			<FormFieldset>
+			<FormFieldset
+				className={ siteIsJetpack && classNames( 'site-settings__has-divider', 'is-top-only' ) }
```

Comment:
No need to call `classNames()` if we have a static set of classes - we can just use:
```
className={ siteIsJetpack && 'site-settings__has-divider is-top-only' }
```